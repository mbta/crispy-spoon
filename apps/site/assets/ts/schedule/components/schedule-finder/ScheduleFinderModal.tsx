import React, { ReactElement } from "react";
import { DirectionId, Route } from "../../../__v3api";
import {
  SimpleStopMap,
  RoutePatternsByDirection,
  ServiceInSelector,
  ScheduleNote as ScheduleNoteType,
  SelectedOrigin
} from "../__schedule";
import Modal from "../../../components/Modal";
import OriginModalContent from "./OriginModalContent";
import ScheduleModalContent from "./ScheduleModalContent";

export type Mode = "origin" | "schedule";

interface Props {
  closeModal: () => void;
  directionChanged?: (direction: DirectionId) => void;
  initialMode: Mode;
  initialDirection: DirectionId;
  initialOrigin: SelectedOrigin;
  originChanged?: (origin: SelectedOrigin) => void;
  route: Route;
  routePatternsByDirection: RoutePatternsByDirection;
  scheduleNote: ScheduleNoteType | null;
  services: ServiceInSelector[];
  stops: SimpleStopMap;
  today: string;
  updateURL: (origin: SelectedOrigin, direction?: DirectionId) => void;
  handleOriginSelectClick: () => void;
}

export default ({
  closeModal,
  directionChanged,
  initialMode,
  initialDirection,
  initialOrigin,
  originChanged,
  route,
  routePatternsByDirection,
  scheduleNote,
  services,
  stops,
  today,
  updateURL,
  handleOriginSelectClick
}: Props): ReactElement => {
  const handleChangeDirection = (newDirection: DirectionId): void => {
    if (directionChanged) directionChanged(newDirection);
    if (originChanged) originChanged(null);
    updateURL(initialOrigin, newDirection);
  };

  const handleChangeOrigin = (newOrigin: SelectedOrigin): void => {
    if (originChanged) originChanged(newOrigin);
    updateURL(newOrigin, initialDirection);
  };

  const originModalContent = (): ReactElement => {
    const origin = initialOrigin;
    const direction = initialDirection;
    return (
      <OriginModalContent
        handleChangeOrigin={handleChangeOrigin}
        selectedOrigin={origin}
        stops={stops[direction] || []}
      />
    );
  };

  const scheduleModalContent = (scheduleOrigin: string): ReactElement => (
    <ScheduleModalContent
      handleChangeDirection={handleChangeDirection}
      handleChangeOrigin={handleChangeOrigin}
      handleOriginSelectClick={handleOriginSelectClick}
      route={route}
      routePatternsByDirection={routePatternsByDirection}
      scheduleNote={scheduleNote}
      selectedDirection={initialDirection}
      selectedOrigin={scheduleOrigin}
      services={services}
      stops={stops}
      today={today}
    />
  );

  const direction = initialDirection;
  const origin = initialOrigin;

  return (
    <Modal
      focusElementId={
        initialMode === "origin" ? "origin-filter" : "modal-close"
      }
      ariaLabel={{
        label:
          initialMode === "origin"
            ? "Choose Origin Stop"
            : `Schedules to ${route.direction_names[direction]}`
      }}
      className={
        initialMode === "origin" ? "schedule-finder__origin-modal" : ""
      }
      closeModal={closeModal}
    >
      {initialMode === "origin" && originModalContent()}
      {initialMode === "schedule" &&
        origin !== null &&
        scheduleModalContent(origin)}
    </Modal>
  );
};
