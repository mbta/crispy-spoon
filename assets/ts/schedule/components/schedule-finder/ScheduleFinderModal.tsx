import React, { ReactElement, useEffect, useState } from "react";
import { DirectionId, Route } from "../../../__v3api";
import {
  SimpleStopMap,
  RoutePatternsByDirection,
  ServiceInSelector,
  SelectedOrigin,
  ScheduleNote
} from "../__schedule";
import Modal from "../../../components/Modal";
import OriginModalContent from "./OriginModalContent";
import ScheduleModalContent from "./ScheduleModalContent";
import { Dispatch } from "redux";
import { useDispatch } from "react-redux";

export type Mode = "origin" | "schedule";

interface Props {
  closeModal: (dispatch: Dispatch) => void;
  directionChanged?: (direction: DirectionId, dispatch: Dispatch) => void;
  initialMode: Mode;
  initialDirection: DirectionId;
  initialOrigin: SelectedOrigin;
  originChanged?: (origin: SelectedOrigin, dispatch: Dispatch) => void;
  route: Route;
  routePatternsByDirection: RoutePatternsByDirection;
  services: ServiceInSelector[];
  stops: SimpleStopMap;
  today: string;
  updateURL: (origin: SelectedOrigin, direction?: DirectionId) => void;
  handleOriginSelectClick: (dispatch: Dispatch) => void;
  scheduleNote: ScheduleNote | null;
}

const ScheduleFinderModal = ({
  closeModal,
  directionChanged,
  initialMode,
  initialDirection,
  initialOrigin,
  originChanged,
  route,
  routePatternsByDirection,
  services,
  stops,
  today,
  updateURL,
  handleOriginSelectClick,
  scheduleNote
}: Props): ReactElement => {
  const [originState, setOriginState] = useState(initialOrigin);
  const dispatch = useDispatch();

  const getStopIdIfInDirection = (newDirection: DirectionId): string | null => {
    const directionDestination = route.direction_destinations[newDirection];
    const stopsForDirection = stops[newDirection];
    const stopsWithoutTerminal = stopsForDirection.filter(
      s => s.name !== directionDestination
    );
    // This needs to return null if the selected origin is the end of the line in the direction
    return stopsWithoutTerminal.find(s => s.id === originState)?.id || null;
  };

  const handleChangeDirection = (newDirection: DirectionId): void => {
    if (directionChanged) directionChanged(newDirection, dispatch);

    // return the stop if it is present in the new direction
    // stops the user from having to reselect an origin on stop change
    const stop = getStopIdIfInDirection(newDirection);

    if (originChanged) originChanged(stop, dispatch);
    updateURL(initialOrigin, newDirection);
  };

  const handleChangeOrigin = (newOrigin: SelectedOrigin): void => {
    setOriginState(newOrigin);
    if (originChanged) originChanged(newOrigin, dispatch);
    updateURL(newOrigin, initialDirection);
  };

  const originModalContent = (): ReactElement => {
    const origin = initialOrigin;
    const direction = initialDirection;
    return (
      <OriginModalContent
        handleChangeOrigin={handleChangeOrigin}
        selectedOrigin={origin}
        stops={stops[direction] || []}
      />
    );
  };

  const scheduleModalContent = (scheduleOrigin: string): ReactElement => (
    <ScheduleModalContent
      handleChangeDirection={handleChangeDirection}
      handleChangeOrigin={handleChangeOrigin}
      handleOriginSelectClick={handleOriginSelectClick}
      route={route}
      routePatternsByDirection={routePatternsByDirection}
      selectedDirection={initialDirection}
      selectedOrigin={scheduleOrigin}
      services={services}
      stops={stops}
      today={today}
      scheduleNote={scheduleNote}
    />
  );

  const direction = initialDirection;
  const origin = initialOrigin;
  const originStop = stops[direction].find(stop => stop.id === origin);

  useEffect(() => {
    setOriginState(initialOrigin);
  }, [initialOrigin]);

  return (
    <Modal
      focusElementId={
        initialMode === "origin" ? "origin-filter" : "modal-close"
      }
      ariaLabel={{
        label:
          initialMode === "origin"
            ? "Choose Origin Stop"
            : `Schedules on the ${route.name} ${
                route.direction_names[direction]
              } to ${route.direction_destinations[direction]}${
                originStop ? ` from ${originStop.name}` : ""
              }`
      }}
      className={
        initialMode === "origin" ? "schedule-finder__origin-modal" : ""
      }
      closeModal={() => closeModal(dispatch)}
    >
      {initialMode === "origin" && originModalContent()}
      {initialMode === "schedule" &&
        origin !== null &&
        scheduleModalContent(origin)}
    </Modal>
  );
};

export default ScheduleFinderModal;
