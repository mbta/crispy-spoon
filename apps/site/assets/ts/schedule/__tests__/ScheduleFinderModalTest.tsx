import React from "react";
import { mount } from "enzyme";
import ScheduleFinderModal, {
  Mode as ModalMode
} from "../components/schedule-finder/ScheduleFinderModal";
import { Route } from "../../__v3api";
import {
  RoutePatternsByDirection,
  SelectedOrigin,
  ServiceInSelector
} from "../components/__schedule";
import * as routePatternsByDirectionData from "./routePatternsByDirectionData.json";

const routePatternsByDirection = routePatternsByDirectionData as RoutePatternsByDirection;

const services: ServiceInSelector[] = [
  {
    valid_days: [1, 2, 3, 4, 5],
    typicality: "typical_service",
    type: "weekday",
    start_date: "2019-07-08",
    removed_dates_notes: {},
    removed_dates: [],
    name: "Weekday",
    id: "BUS319-J-Wdy-02",
    end_date: "2019-08-30",
    description: "Weekday schedule",
    added_dates_notes: {},
    added_dates: [],
    rating_start_date: "2019-06-25",
    rating_end_date: "2019-10-25",
    rating_description: "Test",
    "default_service?": true
  },
  {
    valid_days: [6],
    typicality: "typical_service",
    type: "saturday",
    start_date: "2019-07-13",
    removed_dates_notes: {},
    removed_dates: [],
    name: "Saturday",
    id: "BUS319-K-Sa-02",
    end_date: "2019-08-31",
    description: "Saturday schedule",
    added_dates_notes: {},
    added_dates: [],
    rating_start_date: "2019-06-25",
    rating_end_date: "2019-10-25",
    rating_description: "Test",
    "default_service?": false
  },
  {
    valid_days: [7],
    typicality: "typical_service",
    type: "sunday",
    start_date: "2019-07-14",
    removed_dates_notes: {},
    removed_dates: [],
    name: "Sunday",
    id: "BUS319-L-Su-02",
    end_date: "2019-08-25",
    description: "Sunday schedule",
    added_dates_notes: {},
    added_dates: [],
    rating_start_date: "2019-06-25",
    rating_end_date: "2019-10-25",
    rating_description: "Test",
    "default_service?": false
  }
];

const today = "2019-12-05";

const route: Route = {
  description: "",
  direction_destinations: { 0: "Oak Grove", 1: "Forest Hills" },
  direction_names: { 0: "Inbound", 1: "Outbound" },
  id: "Orange",
  long_name: "Orange Line",
  name: "Orange",
  type: 1
};

const stops = {
  "1": [
    {
      name: "SL",
      id: "741",
      is_closed: false,
      zone: "1"
    },
    {
      name: "Abc",
      id: "123",
      is_closed: false,
      zone: null
    },
    {
      name: "Def",
      id: "456",
      is_closed: false,
      zone: null
    },
    {
      name: "Wellington",
      id: "place-welln",
      is_closed: true,
      zone: null
    }
  ],
  "0": [
    {
      name: "Wellington",
      id: "place-welln",
      is_closed: true,
      zone: null
    },
    {
      name: "Abc",
      id: "123",
      is_closed: false,
      zone: null
    },
    {
      name: "SL",
      id: "741",
      is_closed: false,
      zone: "1"
    }
  ]
};

describe("ScheduleFinderModal", () => {
  const mountComponent = (mode: ModalMode, origin: SelectedOrigin) =>
    mount(
      <ScheduleFinderModal
        closeModal={() => {}}
        initialDirection={0}
        initialMode={"origin"}
        initialOrigin={null}
        route={route}
        routePatternsByDirection={routePatternsByDirection}
        scheduleNote={null}
        services={services}
        stops={stops}
        today={today}
      />
    );

  it("matches snapshot in origin mode", () => {
    const wrapper = mountComponent("origin", null);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("matches snapshot in origin mode with origin selected", () => {
    const wrapper = mountComponent("origin", "place-welln");
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("matches snapshot in schedule mode", () => {
    const wrapper = mountComponent("schedule", "place-welln");
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
