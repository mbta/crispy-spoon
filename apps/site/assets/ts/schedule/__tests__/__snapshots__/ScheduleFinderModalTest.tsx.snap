// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleFinderModal matches snapshot in origin mode 1`] = `
"<Component closeModal={[Function: closeModal]} initialMode=\\"origin\\" route={{...}} routePatternsByDirection={{...}} scheduleNote={{...}} services={{...}} stops={{...}} today=\\"2019-12-05\\" updateURL={[Function: updateURL]} initialDirection={0} initialOrigin={{...}} handleOriginSelectClick={[Function: handleOriginSelectClick]}>
  <WrappedModal focusElementId=\\"origin-filter\\" ariaLabel={{...}} className=\\"schedule-finder__origin-modal\\" closeModal={[Function: closeModal]}>
    <Modal focusElementId=\\"origin-filter\\" ariaLabel={{...}} className=\\"schedule-finder__origin-modal\\" closeModal={[Function: closeModal]} priorPadding=\\"0px\\" paddingRight={0}>
      <ModalContent bodyPadding=\\"0px\\" scrollBarPadding={0} closeText=\\"Close\\" role=\\"dialog\\" ariaLabel={{...}} focusElementId=\\"origin-filter\\" className=\\"schedule-finder__origin-modal\\" closeModal={[Function: closeModal]}>
        <Portal containerInfo={{...}}>
          <aside aria-modal={true} aria-label=\\"Choose Origin Stop\\" role=\\"dialog\\" id=\\"modal-cover\\" className=\\"c-modal__cover\\" tabIndex={-1} onKeyDown={[Function: onKeyDown]} onClick={[Function: onClick]}>
            <div className=\\"c-modal schedule-finder__origin-modal\\">
              <button id=\\"modal-close\\" className=\\"btn btn-secondary c-modal__close\\" onClick={[Function: onClick]} type=\\"button\\">
                Close
              </button>
              <div className=\\"c-modal__content\\">
                <OriginModalContent handleChangeOrigin={[Function: handleChangeOrigin]} selectedOrigin={{...}} stops={{...}}>
                  <br />
                  <h3 className=\\"schedule-finder__origin-text\\">
                    Choose an origin stop
                  </h3>
                  <SearchBox id=\\"origin-filter\\" labelText=\\"Choose an origin stop\\" className=\\"schedule-finder__origin-search-container\\" placeholder=\\"Filter stops and stations\\" onChange={[Function: bound dispatchAction]}>
                    <div className=\\"schedule-finder__origin-search-container c-form__input-container\\">
                      <label htmlFor=\\"origin-filter\\" className=\\"sr-only\\">
                        Choose an origin stop
                      </label>
                      <input className=\\"c-form__input\\" id=\\"origin-filter\\" type=\\"text\\" onChange={[Function: onChange]} value=\\"\\" placeholder=\\"Filter stops and stations\\" />
                      <button className=\\"c-form__submit-btn\\" aria-label=\\"search\\" type=\\"button\\" onClick={[Function: onClick]}>
                        <span className=\\"notranslate c-svg__icon-search-reverse-default\\" aria-hidden=\\"false\\" dangerouslySetInnerHTML={{...}} />
                      </button>
                    </div>
                  </SearchBox>
                  <p className=\\"schedule-finder__origin-text\\">
                    Select from the list below.
                  </p>
                  <div className=\\"schedule-finder__origin-list\\">
                    <OriginListItem stop={{...}} changeOrigin={[Function: handleChangeOrigin]} selectedOrigin={{...}} lastStop={{...}}>
                      <div tabIndex={0} role=\\"button\\" className=\\"schedule-finder__origin-list-item  disabled\\" onClick={[Function: onClick]} onKeyUp={[Function: onKeyUp]}>
                        <div className=\\"schedule-finder__origin-list-leftpad\\">
                           
                        </div>
                        Wellington
                         
                      </div>
                    </OriginListItem>
                    <OriginListItem stop={{...}} changeOrigin={[Function: handleChangeOrigin]} selectedOrigin={{...}} lastStop={{...}}>
                      <div tabIndex={0} role=\\"button\\" className=\\"schedule-finder__origin-list-item  \\" onClick={[Function: onClick]} onKeyUp={[Function: onKeyUp]}>
                        <div className=\\"schedule-finder__origin-list-leftpad\\">
                           
                        </div>
                        Abc
                         
                      </div>
                    </OriginListItem>
                    <OriginListItem stop={{...}} changeOrigin={[Function: handleChangeOrigin]} selectedOrigin={{...}} lastStop={{...}}>
                      <div tabIndex={0} role=\\"button\\" className=\\"schedule-finder__origin-list-item  disabled\\" onClick={[Function: onClick]} onKeyUp={[Function: onKeyUp]}>
                        <div className=\\"schedule-finder__origin-list-leftpad\\">
                           
                        </div>
                        SL
                         
                        <span className=\\"schedule-finder__zone\\">
                          Zone 
                          1
                        </span>
                      </div>
                    </OriginListItem>
                  </div>
                </OriginModalContent>
              </div>
            </div>
          </aside>
        </Portal>
      </ModalContent>
    </Modal>
  </WrappedModal>
</Component>"
`;

exports[`ScheduleFinderModal matches snapshot in origin mode with origin selected 1`] = `
"<Component closeModal={[Function: closeModal]} initialMode=\\"origin\\" route={{...}} routePatternsByDirection={{...}} scheduleNote={{...}} services={{...}} stops={{...}} today=\\"2019-12-05\\" updateURL={[Function: updateURL]} initialDirection={0} initialOrigin={{...}} handleOriginSelectClick={[Function: handleOriginSelectClick]}>
  <WrappedModal focusElementId=\\"origin-filter\\" ariaLabel={{...}} className=\\"schedule-finder__origin-modal\\" closeModal={[Function: closeModal]}>
    <Modal focusElementId=\\"origin-filter\\" ariaLabel={{...}} className=\\"schedule-finder__origin-modal\\" closeModal={[Function: closeModal]} priorPadding=\\"0px\\" paddingRight={0}>
      <ModalContent bodyPadding=\\"0px\\" scrollBarPadding={0} closeText=\\"Close\\" role=\\"dialog\\" ariaLabel={{...}} focusElementId=\\"origin-filter\\" className=\\"schedule-finder__origin-modal\\" closeModal={[Function: closeModal]}>
        <Portal containerInfo={{...}}>
          <aside aria-modal={true} aria-label=\\"Choose Origin Stop\\" role=\\"dialog\\" id=\\"modal-cover\\" className=\\"c-modal__cover\\" tabIndex={-1} onKeyDown={[Function: onKeyDown]} onClick={[Function: onClick]}>
            <div className=\\"c-modal schedule-finder__origin-modal\\">
              <button id=\\"modal-close\\" className=\\"btn btn-secondary c-modal__close\\" onClick={[Function: onClick]} type=\\"button\\">
                Close
              </button>
              <div className=\\"c-modal__content\\">
                <OriginModalContent handleChangeOrigin={[Function: handleChangeOrigin]} selectedOrigin={{...}} stops={{...}}>
                  <br />
                  <h3 className=\\"schedule-finder__origin-text\\">
                    Choose an origin stop
                  </h3>
                  <SearchBox id=\\"origin-filter\\" labelText=\\"Choose an origin stop\\" className=\\"schedule-finder__origin-search-container\\" placeholder=\\"Filter stops and stations\\" onChange={[Function: bound dispatchAction]}>
                    <div className=\\"schedule-finder__origin-search-container c-form__input-container\\">
                      <label htmlFor=\\"origin-filter\\" className=\\"sr-only\\">
                        Choose an origin stop
                      </label>
                      <input className=\\"c-form__input\\" id=\\"origin-filter\\" type=\\"text\\" onChange={[Function: onChange]} value=\\"\\" placeholder=\\"Filter stops and stations\\" />
                      <button className=\\"c-form__submit-btn\\" aria-label=\\"search\\" type=\\"button\\" onClick={[Function: onClick]}>
                        <span className=\\"notranslate c-svg__icon-search-reverse-default\\" aria-hidden=\\"false\\" dangerouslySetInnerHTML={{...}} />
                      </button>
                    </div>
                  </SearchBox>
                  <p className=\\"schedule-finder__origin-text\\">
                    Select from the list below.
                  </p>
                  <div className=\\"schedule-finder__origin-list\\">
                    <OriginListItem stop={{...}} changeOrigin={[Function: handleChangeOrigin]} selectedOrigin={{...}} lastStop={{...}}>
                      <div tabIndex={0} role=\\"button\\" className=\\"schedule-finder__origin-list-item  disabled\\" onClick={[Function: onClick]} onKeyUp={[Function: onKeyUp]}>
                        <div className=\\"schedule-finder__origin-list-leftpad\\">
                           
                        </div>
                        Wellington
                         
                      </div>
                    </OriginListItem>
                    <OriginListItem stop={{...}} changeOrigin={[Function: handleChangeOrigin]} selectedOrigin={{...}} lastStop={{...}}>
                      <div tabIndex={0} role=\\"button\\" className=\\"schedule-finder__origin-list-item  \\" onClick={[Function: onClick]} onKeyUp={[Function: onKeyUp]}>
                        <div className=\\"schedule-finder__origin-list-leftpad\\">
                           
                        </div>
                        Abc
                         
                      </div>
                    </OriginListItem>
                    <OriginListItem stop={{...}} changeOrigin={[Function: handleChangeOrigin]} selectedOrigin={{...}} lastStop={{...}}>
                      <div tabIndex={0} role=\\"button\\" className=\\"schedule-finder__origin-list-item  disabled\\" onClick={[Function: onClick]} onKeyUp={[Function: onKeyUp]}>
                        <div className=\\"schedule-finder__origin-list-leftpad\\">
                           
                        </div>
                        SL
                         
                        <span className=\\"schedule-finder__zone\\">
                          Zone 
                          1
                        </span>
                      </div>
                    </OriginListItem>
                  </div>
                </OriginModalContent>
              </div>
            </div>
          </aside>
        </Portal>
      </ModalContent>
    </Modal>
  </WrappedModal>
</Component>"
`;

exports[`ScheduleFinderModal matches snapshot in schedule mode 1`] = `
"<Component closeModal={[Function: closeModal]} initialMode=\\"origin\\" route={{...}} routePatternsByDirection={{...}} scheduleNote={{...}} services={{...}} stops={{...}} today=\\"2019-12-05\\" updateURL={[Function: updateURL]} initialDirection={0} initialOrigin={{...}} handleOriginSelectClick={[Function: handleOriginSelectClick]}>
  <WrappedModal focusElementId=\\"origin-filter\\" ariaLabel={{...}} className=\\"schedule-finder__origin-modal\\" closeModal={[Function: closeModal]}>
    <Modal focusElementId=\\"origin-filter\\" ariaLabel={{...}} className=\\"schedule-finder__origin-modal\\" closeModal={[Function: closeModal]} priorPadding=\\"0px\\" paddingRight={0}>
      <ModalContent bodyPadding=\\"0px\\" scrollBarPadding={0} closeText=\\"Close\\" role=\\"dialog\\" ariaLabel={{...}} focusElementId=\\"origin-filter\\" className=\\"schedule-finder__origin-modal\\" closeModal={[Function: closeModal]}>
        <Portal containerInfo={{...}}>
          <aside aria-modal={true} aria-label=\\"Choose Origin Stop\\" role=\\"dialog\\" id=\\"modal-cover\\" className=\\"c-modal__cover\\" tabIndex={-1} onKeyDown={[Function: onKeyDown]} onClick={[Function: onClick]}>
            <div className=\\"c-modal schedule-finder__origin-modal\\">
              <button id=\\"modal-close\\" className=\\"btn btn-secondary c-modal__close\\" onClick={[Function: onClick]} type=\\"button\\">
                Close
              </button>
              <div className=\\"c-modal__content\\">
                <OriginModalContent handleChangeOrigin={[Function: handleChangeOrigin]} selectedOrigin={{...}} stops={{...}}>
                  <br />
                  <h3 className=\\"schedule-finder__origin-text\\">
                    Choose an origin stop
                  </h3>
                  <SearchBox id=\\"origin-filter\\" labelText=\\"Choose an origin stop\\" className=\\"schedule-finder__origin-search-container\\" placeholder=\\"Filter stops and stations\\" onChange={[Function: bound dispatchAction]}>
                    <div className=\\"schedule-finder__origin-search-container c-form__input-container\\">
                      <label htmlFor=\\"origin-filter\\" className=\\"sr-only\\">
                        Choose an origin stop
                      </label>
                      <input className=\\"c-form__input\\" id=\\"origin-filter\\" type=\\"text\\" onChange={[Function: onChange]} value=\\"\\" placeholder=\\"Filter stops and stations\\" />
                      <button className=\\"c-form__submit-btn\\" aria-label=\\"search\\" type=\\"button\\" onClick={[Function: onClick]}>
                        <span className=\\"notranslate c-svg__icon-search-reverse-default\\" aria-hidden=\\"false\\" dangerouslySetInnerHTML={{...}} />
                      </button>
                    </div>
                  </SearchBox>
                  <p className=\\"schedule-finder__origin-text\\">
                    Select from the list below.
                  </p>
                  <div className=\\"schedule-finder__origin-list\\">
                    <OriginListItem stop={{...}} changeOrigin={[Function: handleChangeOrigin]} selectedOrigin={{...}} lastStop={{...}}>
                      <div tabIndex={0} role=\\"button\\" className=\\"schedule-finder__origin-list-item  disabled\\" onClick={[Function: onClick]} onKeyUp={[Function: onKeyUp]}>
                        <div className=\\"schedule-finder__origin-list-leftpad\\">
                           
                        </div>
                        Wellington
                         
                      </div>
                    </OriginListItem>
                    <OriginListItem stop={{...}} changeOrigin={[Function: handleChangeOrigin]} selectedOrigin={{...}} lastStop={{...}}>
                      <div tabIndex={0} role=\\"button\\" className=\\"schedule-finder__origin-list-item  \\" onClick={[Function: onClick]} onKeyUp={[Function: onKeyUp]}>
                        <div className=\\"schedule-finder__origin-list-leftpad\\">
                           
                        </div>
                        Abc
                         
                      </div>
                    </OriginListItem>
                    <OriginListItem stop={{...}} changeOrigin={[Function: handleChangeOrigin]} selectedOrigin={{...}} lastStop={{...}}>
                      <div tabIndex={0} role=\\"button\\" className=\\"schedule-finder__origin-list-item  disabled\\" onClick={[Function: onClick]} onKeyUp={[Function: onKeyUp]}>
                        <div className=\\"schedule-finder__origin-list-leftpad\\">
                           
                        </div>
                        SL
                         
                        <span className=\\"schedule-finder__zone\\">
                          Zone 
                          1
                        </span>
                      </div>
                    </OriginListItem>
                  </div>
                </OriginModalContent>
              </div>
            </div>
          </aside>
        </Portal>
      </ModalContent>
    </Modal>
  </WrappedModal>
</Component>"
`;
